src/strategy/mod.rs

```diff
@@ -6,4 +8,26 @@ mod jitter;
 pub use self::fixed_interval::FixedInterval;
 pub use self::exponential_backoff::ExponentialBackoff;
 pub use self::fibonacci_backoff::FibonacciBackoff;
-pub use self::jitter::jitter;
\ No newline at end of file
+pub use self::jitter::jitter;
+
+/// Internally used to specify a maximum delay for certain backoff strategies.
+#[derive(Debug, Clone)]
+struct MaxDelay {
+    /// Maximum desired delay
+    duration: Duration,
+    /// `true` if the maximum delay has been reached by the given backoff strategy.
+    is_reached: bool,
+}
+
+impl MaxDelay {
+    fn new(max_delay: Option<Duration>) -> Option<MaxDelay> {
```

Comment:
Having `Option` in both input and output types seems weird here. This could either be simplified into `fn new(max_delay: Duration) -> MaxDelay`, and the code that uses it would just go `max_delay.map(MaxDelay::new)`, or you change `MaxDelay` to have `duration: Option<Duration>` so that this signature becomes `fn new(max_delay: Option<Duration>) -> MaxDelay`.